Add hotdog OCI hooks

arnaldo2792 opened the pull request:
**Issue number:**
N / A


**Description of changes:**
`hotdog` provides oci hooks to apply the hotpatch for Apache log4j2, to containers running JVM machines

Remaining tasks:

- [x] Change the sources to the public hotdog repo
- [x] Update cache

**Testing done:**
I built it with my local copy of the sources.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

arnaldo2792 committed:
packages: add hotdog

hotdog provides oci hooks to apply the hotpatch for Apache log4j2, to
containers running JVM machines

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

bcressey commented on packages/release/release.spec:
```diff
@@ -86,6 +86,7 @@ Requires: %{_cross_os}selinux-policy
 Requires: %{_cross_os}systemd
 Requires: %{_cross_os}util-linux
 Requires: %{_cross_os}wicked
+Requires: %{_cross_os}hotdog
```
nit: should go in alphabetical order

bcressey commented on packages/hotdog/hotdog.spec:
```diff
@@ -0,0 +1,68 @@
+cp GOPATH/src/golang.org/x/sys/LICENSE LICENSE.golang-sys
+
+%build
+# Set CGO_ENABLED=0 to statically link hotdog-hotpath, since it runs inside containers that
+# may not have the glibc version used to compile it
+# Set `GO111MODULE=off` to force golang to look for the dependencies in the GOPATH
+export GOPATH=$PWD/GOPATH
+CGO_ENABLED=0 GO111MODULE=off go build -buildmode=pie -ldflags "-linkmode=external" -o hotdog-hotpatch ./cmd/hotdog-hotpatch
```
I don't think we want to force the external linker here if we're not also enabling cgo in order to link with our libc. I'd just use the same command from the hotdog Makefile.

samuelkarp commented on packages/hotdog/Cargo.toml:
```diff
@@ -0,0 +1,25 @@
+
+[[package.metadata.build-package.external-files]]
+# TODO: fix
+url = "http://local.fix.this/hotdog-main.tar.gz"
```
https://github.com/bottlerocket-os/hotdog

We've got public sources now :smile:

arnaldo2792 commented:
Forced push includes:

- Update sources to reference public repository
- Ordering nit in release.spec
- Added LICENSE file from `hotdog` sources
- Updated license fro `hotdog`
- Used the upstream compile command for `hotdog-hotpatch`

samuelkarp reviewed (approved):
Code LGTM.  We need to get this tested in combination with the other PRs too.

bcressey commented on packages/hotdog/Cargo.toml:
```diff
@@ -0,0 +1,24 @@
+[lib]
+path = "pkg.rs"
+
+[[package.metadata.build-package.external-files]]
+url = "https://github.com/bottlerocket-os/hotdog/archive/c212cfb564850b48aa6008563074a329cee041a1/hotdog-c212cfb.tar.gz"
+sha512 = "68bbd4c4fda72065eae995fe75f0f8d0a30228476051906da4bce3f66352138356bb4dd4fd8e0aa6f5822388f9c7dc04e85fa35038e91be294e4fdd4a6e970bc"
```
We want https://github.com/bottlerocket-os/hotdog/commit/f300fb78 for the verbosity fix.

bcressey commented on packages/hotdog/hotdog.spec:
```diff
@@ -0,0 +1,79 @@
+# The oci hooks commands can be compiled as we usually compile golang packages
+%cross_go_configure %{goimport}
+for cmd in hotdog-cc-hook hotdog-poststart-hook; do
+  GO111MODULE=off go build -buildmode=pie -ldflags "-linkmode=external" -o $cmd ./cmd/$cmd
```
This probably needs to change for the newer SDK:
```suggestion
  GO111MODULE=off go build -buildmode=pie -ldflags "${GOLDFLAGS}" -o $cmd ./cmd/$cmd
```

arnaldo2792 commented:
Forced push includes:

- Updated the version in hotdog
- Used the new flags in build
